client/src/annoMatrix/crossfilter.ts

```diff
@@ -9,38 +9,49 @@ AnnoMatrix stay in sync:
 */
 import Crossfilter from "../util/typedCrossfilter";
 import { _getColumnSchema } from "./schema";
-
-// @ts-expect-error ts-migrate(7006) FIXME: Parameter 'field' implicitly has an 'any' type.
-function _dimensionNameFromDf(field, df) {
+import {
+  AnnotationColumn,
+  Field,
+  LayoutColumn,
+} from "../common/types/entities";
+import AnnoMatrix, { ObsColumnValue } from "./annoMatrix";
+import { Dataframe } from "../util/dataframe";
+import { Query } from "./query";
+
+type ObsDimensionParams =
+  | [string, Float32Array | undefined, Float32Array | undefined]
+  | [string, string[]]
+  | [string, string[], Float32ArrayConstructor]
+  | [string, string[], Int32ArrayConstructor]
+  | undefined;
+
+function _dimensionNameFromDf(field: Field, df: Dataframe): string {
   const colNames = df.colIndex.labels();
   return _dimensionName(field, colNames);
 }
 
-// @ts-expect-error ts-migrate(7006) FIXME: Parameter 'field' implicitly has an 'any' type.
-function _dimensionName(field, colNames) {
+function _dimensionName(field: Field, colNames: string | string[]): string {
```

Comment:
colNames can also be a Int32Array (on L29, df.colIndex.labels() could return Int32Index or Array<string>)

Reply:
Updated [crossfilter.ts::_dimensionName](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L35).

Reply:
Dataframe has types for this now -- should be `LabelType | LabelArray`

See new definition for LabelIndex.labels() in PR #2382

Reply:
Updated [crossfilter.ts::_dimensionName](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/annoMatrix/crossfilter.ts#L41).